lib/ex_doc/markdown_processor.ex

```diff
@@ -0,0 +1,90 @@
+defmodule ExDoc.MarkdownProcessor do
+  @moduledoc """
+  ExDoc is compatible with markdown processors that implement this behaviour.
+
+  If you want to write a custom markdown processor that plays well with ExDoc,
+  you must implement this behaviour in your module.
+  If you're not interested in writing a custom markdown processor for use with ExDoc,
+  you can safely ignore this module.
+  """
+
+  @doc """
+  Converts markdown into HTML.
+  """
+  @callback to_html(String.t, Keyword.t) :: String.t
+
+  @doc """
+  Assets specific to the markdown implementation.
+
+  This callback takes the documentation format (`:html` or `epub`) as an argument
+  and must return a list of pairs of the form: `{basename, content}` where:
+
+  - `basename` relative path that will be written inside the `doc/` directory.
```

Comment:
Please use `*` for lists and indent them with two spaces.

Reply:
Fixed.